tests/console/systemd_testsuite.pm

```diff
@@ -0,0 +1,53 @@
+# SUSE's systemd tests
+#
+# Copyright © 2017 SUSE LLC
+#
+# Copying and distribution of this file, with or without modification,
+# are permitted in any medium without royalty provided the copyright
+# notice and this notice are preserved.  This file is offered as-is,
+# without any warranty.
+
+# Summary: Run testsuite included in systemd sources
+# Maintainer: Thomas Blume <tblume@suse.com>
+
+use base "consoletest";
+use warnings;
+use strict;
+use testapi;
+use utils;
+
+sub run() {
+    # install systemd testsuite
+    select_console 'root-console';
+    if (check_var('VERSION', '12-SP2')) {
+        zypper_call('ar http://download.suse.de/ibs/QA:/SLE12SP2/update/ systemd-testrepo');
+    }
+
+    elsif (check_var('VERSION', '12-SP3')) {
+        zypper_call('ar http://download.suse.de/ibs/QA:/SLE12SP3/update/ systemd-testrepo');
+    }
+    else {
```

Comment:
Hm, we might have a misunderstanding. You are talking about publishing of the package.
AFAIK, that happens after it has been submitted into one of the official projects.
I'm talking about a simple build of the package, maybe a local build.
I don't want to run the testsuite at each package build. That is overkill.
Furthermore, we don't do a package update for each particular bug.
Since systemd has a high number of bug reports, we would keep the maintenance team busy with only processing systemd updates, if we would do so.
We rather collect a number of patches and do an submission when enough have been assembled.
At that time, a testsuite run makes sense and that is what we do, as stated in my previous comment.

I don't want to shift my testing workload to openQA, but we as maintainers were asked to provide QA tests for our packages.
And I saw no problem with this since it is an automated procedure and you only need to get active when a test fails. Concerning the investigation overhead, a package testsuite is for a certain package and if it fails you can directly open a bug to the package maintainer. What else would you need to investigate then?

Finally, at least the systemd testsuite is almost pure system testing. A failure that you capture therewith can also happen during boot of your test system. At that stage it is very often not obvious that systemd is the culprit, so your investigateon overhead would be higher than if the problem was discovered directly with the systemd testsuite.

However, I'm open for altenative ways to do automated package testing. I took a look at your multibuild package test and this seems to be pretty different from openQA test code.
If this is the way to go, you are welcome to use my test package provided at ibs/QA:, but please don't expect me to add tests there myself.
I already spend a lot of time to learn openQA test creation and don't intend to spend more time to learn another test submission procedure. :(